common/src/services/crypto.service.ts

```diff
@@ -336,74 +325,71 @@ export class CryptoService implements CryptoServiceAbstraction {
     }
 
     async hasKey(): Promise<boolean> {
-        return this.hasKeyInMemory() || await this.hasKeyStored('auto') || await this.hasKeyStored('biometric');
+        return await this.hasKeyInMemory() || await this.hasKeyStored(KeySuffixOptions.Auto) || await this.hasKeyStored(KeySuffixOptions.Biometric);
     }
 
-    hasKeyInMemory(): boolean {
-        return this.key != null;
+    async hasKeyInMemory(userId?: string): Promise<boolean> {
+        return await this.stateService.getCryptoMasterKey({ userId: userId }) != null;
     }
 
-    hasKeyStored(keySuffix: KeySuffixOptions): Promise<boolean> {
-        return this.secureStorageService.has(Keys.key, { keySuffix: keySuffix });
+    async hasKeyStored(keySuffix: KeySuffixOptions, userId?: string): Promise<boolean> {
+        return await this.stateService.getCryptoMasterKeyB64({ keySuffix: keySuffix, userId: userId }) != null;
```

Comment:
Why are we using B64 in some places and not b64 in others?

Reply:
🤷‍♂️  I don't have a great understanding of how all the keys interact with each other, I just know this is how it was was working before. The B64 version is stored in secure storage and the full key is stored in memory, and both are used throughout the code.